client/state/posts/utils.js

```diff
@@ -241,6 +243,33 @@ export function normalizeTermsForApi( post ) {
 }
 
 /**
+ * Returns truthy if local terms object is different than API response
+ *
+ * @param  {Object}  localTermEdits local state of term edits
+ * @param  {Object}  savedTerms     term object returned from API POST
+ * @return {Boolean}                are there differences in local edits vs saved terms
+ */
+export function hasTermEditDifferences( localTermEdits, savedTerms ) {
+	if ( ! localTermEdits || ! savedTerms ) {
+		return false;
+	}
+
+	return !! find( localTermEdits, ( terms, taxonomy ) => {
```

Comment:
Minor: [`_.some`](https://lodash.com/docs/4.17.4#some) might be more precise to what you're trying to achieve (returns boolean rather than the found value).